sdk/eventhub/event-hubs/src/partitionReceiver.ts

```diff
@@ -351,28 +351,43 @@ export function waitForEvents(
     receivedAfterWait,
     receivedAlready,
   } = options;
+
+  if (queue.length >= maxEventCount) {
+    return Promise.resolve().then(receivedAlready);
+  }
```

Comment:
Is this just a matter of style or is there a need to move it?

Reply:
We don't need to deal with abort signal if we are going to just return already received messagses?

Reply:
I am not sure if I understand your concern 100% but I believe the abort signal logic in the `finally` call applies only to the race promise and not to this promise in the old code, no?

Reply:
that's right because it is not affected by abort signal so I moved it up and return early. I can move it back if you prefer that way.

Reply:
I still don't see the connection. The `finally` call was not chained to this promise at all so it doesn't matter.

Reply:
Discussed offline with @deyaaeldeen.  It's a small optimization that for the `queue.length >= maxEventCount` case there's no need to create the aborter and updatedOptions object.